app/models/manageiq/providers/openstack/cloud_manager/provision_workflow.rb

```diff
@@ -23,6 +23,36 @@ def validate_cloud_network(field, values, dlg, fld, value)
     validate_placement(field, values, dlg, fld, value)
   end
 
+  def set_request_values(values)
+    values[:volumes] = prepare_volumes_fields(values)
+    super
+  end
+
+  def prepare_volumes_fields(values)
+    # the provision dialog doesn't handle arrays,
```

Comment:
@bdunne I feel like that refactor might be better suited for a future PR, as it sounds like an inherent issue with the dialog and not one introduced by this PR.